algorithms/src/std_algorithms/impl/Kokkos_RandomAccessIterator.hpp

```diff
@@ -176,15 +344,14 @@ class RandomAccessIterator< ::Kokkos::View<DataType, Args...> > {
   KOKKOS_FUNCTION
   reference operator*() const { return m_view(m_current_index); }
 
-  KOKKOS_FUNCTION
-  view_type view() const { return m_view; }
-
  private:
-  view_type m_view;
+  ::Kokkos::View<DataType, typename view_type::memory_space,
+                 Kokkos::MemoryUnmanaged>
+      m_view;
```

Comment:
Would the following be more appropriate ?

```suggestion
  using view_um_type = Kokkos::View<
        typename view_type::traits::value_type,
        typename view_type::traits::array_layout,
        typename view_type::traits::device_type,
        typename view_type::traits::hooks_policy,
        Kokkos::MemoryTraits<Kokkos::Unmanaged>
    >;
  view_um_t m_view;
```

Reply:
(basically you'd copy all traits but the memory one)